gnovm/cmd/gno/test.go

```diff
@@ -227,44 +227,52 @@ func execTest(cfg *testCfg, args []string, io commands.IO) error {
 			io.ErrPrintfln("?       %s \t[no test files]", pkg.Dir)
 			continue
 		}
-		// Determine gnoPkgPath by reading gno.mod
-		modfile, _ := gnomod.ParseAt(pkg.Dir)
-		gnoPkgPath, ok := determinePkgPath(modfile, pkg.Dir, cfg.rootDir)
+
+		// Determine gnoPkgPath by reading gno.mod from disk.
+		// TODO: Have ReadMemPackage handle it.
+		modfile, _ := gnomod.ParseDir(pkg.Dir)
+		gnoPkgPath, ok := determinePkgPath(modfile, pkg.Dir, cmd.rootDir)
 		if !ok {
 			io.ErrPrintfln("--- WARNING: unable to read package path from gno.mod or gno root directory; try creating a gno.mod file")
 		}
 
+		// Read MemPackage and lint/typecheck/format.
+		// (gno.mod will be read again).
+		var errsLint, errsTest error
 		memPkg := gno.MustReadMemPackage(pkg.Dir, gnoPkgPath)
-
-		var hasError bool
-
 		startedAt := time.Now()
-		runtimeError := catchRuntimeError(pkg.Dir, gnoPkgPath, io.Err(), func() {
+		didPanic := catchPanic(pkg.Dir, gnoPkgPath, io.Err(), func() {
 			if modfile == nil || !modfile.Draft {
-				foundErr, lintErr := lintTypeCheck(io, pkg.Dir, memPkg, opts.TestStore)
-				if lintErr != nil {
-					io.ErrPrintln(lintErr)
-					hasError = true
-				} else if foundErr {
-					hasError = true
+				_, _, _, errs := lintTypeCheck(io, pkg.Dir, memPkg, opts.TestStore)
+				if errs != nil {
+					errsLint = errs
+					io.ErrPrintln(errs)
 				}
-			} else if cfg.verbose {
+			} else if cmd.verbose {
 				io.ErrPrintfln("%s: module is draft, skipping type check", gnoPkgPath)
 			}
-			err = test.Test(memPkg, pkg.Dir, opts)
+			errs := test.Test(memPkg, pkg.Dir, opts)
+			if errs != nil {
+				errsTest = errs
+				io.ErrPrintln(errs)
+			}
 		})
-		hasError = hasError || runtimeError
 
+		// Print status with duration.
 		duration := time.Since(startedAt)
 		dstr := fmtDuration(duration)
-
-		if hasError || err != nil {
-			if err != nil {
-				io.ErrPrintfln("%s: test pkg: %v", pkg.Dir, err)
+		if didPanic || errsLint != nil || errsTest != nil {
+			if errsLint != nil {
```

Comment:
I'm considering whether it makes sense to keep a standalone `gno lint` command instead of integrating linting into `gno test`. What do you think?

By the way, I was also thinking about making `gno lint` return an error if a mempackage lacks at least one test/filetest file. It's acceptable not to have these if you `addpkg`, but it's not acceptable for the CLI and CI systems to lack at least one test file (even if it's empty) for a gno package. I believe this approach aligns well with our ecosystem. Wdyt?

Reply:
For my first point, I believe it's acceptable to have a more comprehensive (full) linting process in `gno lint` while keeping minimal (type check only) linting minimal here. 

Alternatively, we could consider type check linting as part of `gno test`, making `gno lint` encompass everything else.